src/compiler/compile/Component.ts

```diff
@@ -191,27 +191,33 @@ export default class Component {
 		this.stylesheet.warn_on_unused_selectors(this);
 	}
 
-	add_var(variable: Var, add_to_lookup = true) {
+	add_var(node: Node, variable: Var, add_to_lookup = true) {
 		this.vars.push(variable);
 
 		if (add_to_lookup) {
+			if (this.var_lookup.has(variable.name)) {
+				const existsVar = this.var_lookup.get(variable.name);
+				if (existsVar.module && existsVar.imported) {
+					this.error(node as any, compiler_errors.illegal_variable_declaration);
+				}
+			}
 			this.var_lookup.set(variable.name, variable);
 		}
 	}
```

Comment:
Adding a new `node: Node` param seems to be affecting a lot of other code, but seems like it's needed for the validation to work. If this is fine by the others, it's an approval from me.

Reply:
Thank you for the comment!
Yes. I also want to avoid adding this argument.
But I couldn't find an alternative way.
If someone has an idea, I will update it according to it.